### PR TITLE
fix(auth): Add interfaces for UserProviderRequest and Metadata used in UserImportRecord

### PR DESCRIPTION
**Pull request by MathBunny**
I think there is an inconsistency with the defined `auth.d.ts` file [here](https://github.com/firebase/firebase-admin-node/blob/master/src/auth.d.ts#L802). This aims to add the interface to be consistent and then have an array of `UserProviderRequest`.

Similar change applies also for [metadata](https://github.com/firebase/firebase-admin-node/blob/master/src/auth.d.ts#L696).

Not sure if this is the best approach.

### CODE REVIEW

File: src/auth/user-import-builder.ts [hunk 1]
```diff
@@ -47,6 +47,20 @@ interface SecondFactor {
...
+  email?: string;
+  phoneNumber?: string;
+  photoURL?: string;
+  providerId: string;
+}
+
+interface UserMetadataRequest {
+  lastSignInTime?: string;
+  creationTime?: string;
+  lastRefreshTime?: string|null;
```
Comment (bojeil-google) [inline src/auth/user-import-builder.ts:62]: I am not sure `lastRefreshTime` is supported for import operations. Please remove this one for now.

Comment (MathBunny) [reply src/auth/user-import-builder.ts]: Thanks -- are you referring to the specific field or should I remove the entire interface?

Comment (bojeil-google) [reply src/auth/user-import-builder.ts]: Just this specific field.

Comment (hiranya911) [reply src/auth/user-import-builder.ts]: @bojeil-google we have the following in our typings for `UserMetadataRequest`:

```
lastRefreshTime?: string|null;
```

Should we remove that too?

**Pull request merged**